alf/environments/mario_wrappers.py

```diff
@@ -0,0 +1,246 @@
+# Copyright (c) 2019 Horizon Robotics. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import deque
+import itertools
+from copy import copy
+import numpy as np
+from PIL import Image
+import gym
+from gym import spaces
+
+
+# See https://github.com/openai/large-scale-curiosity/blob/ \
+#  0c3d179fd61ee46233199d0891c40fbe7964d3aa/wrappers.py#L155-L238
+
+class MarioXReward(gym.Wrapper):
+    """
+    Wrap mario environment and use X-axis coordinate increment as reward
+
+    ```
+        max_x = 0 if initial or upgrade_to_new_level
+        current_x = xscrollHi * 256 + xscrollLo
+        reward = current_x - max_x if current_x > max_x else 0
```

Comment:
So it's not purely based on the intrinsic reward. Do you know what exactly current_x represents? And what will the result if not using it? What

Reply:
1. It's base on intrinsic and extrinsic reward ,  i am try using experiments with only intrinsic reward ,  but result is not good yet

2. current_x mean how far the role go forward ,  `xscrollHi`  for  `page num` ,  `xscrollLo`  for `x position in current page`   and page size is `256`